lib/p2p/Parser.ts

```diff
@@ -60,20 +60,30 @@ interface Parser {
 
 /** Protocol packet parser */
 class Parser extends EventEmitter {
+  private pending = '';
+
+  constructor(private delimiter: string) {
+    super();
+  }
+
   public feed = (data: Buffer): void => {
-    const dataStr = data.toString();
-    const packetLines = dataStr.split('\r\n');
-    packetLines.forEach((packetStr) => {
-      if (!packetStr) {
-        return;
-      }
-      try {
-        const packet = fromRaw(packetStr);
-        this.emit('packet', packet);
-      } catch (err) {
-        this.emit('error', err);
-      }
-    });
+    this.pending += data.toString();
+    const index = this.pending.indexOf(this.delimiter);
+    if (index > -1) {
+      this.parsePacket(this.pending.slice(0, index));
+      const next = this.pending.slice(index + this.delimiter.length);
+      this.pending = '';
+      this.feed(Buffer.from(next));
```

Comment:
We're converting a buffer to a string and then back to a buffer here, I'm thinking it might be better to make `feed()` accept a string parameter and do `toString()` from where we call the method in Peer.ts. It would also make the tests a little cleaner, since we have `Buffer.from` calls in multiple places that we wouldn't need if `feed` took a string.

Reply:
I left it as is because I thought that maybe at some point we'll want to read the bytes array and not the string. But I can do what you suggester for now.